doc/src/contributing/new-contributors-guide/writing-tests.md

```diff
@@ -0,0 +1,1078 @@
+# Writing Tests
+
+The most important thing for a mathematical library like SymPy is correctness.
+Functions should never return mathematically incorrect results. Correctness is
+always the top concern, even if it comes at the cost of things like
+performance or modularity.
+
+Consequently, all functionality in SymPy is tested extensively. This guide
+goes over how tests in SymPy are written.
+
+## Testing Policies
+
+In order to ensure the high standard of correctness, SymPy has the following
+rules that apply to all pull requests:
+
+1. All new functionality must be tested. Tests should aim to cover all
+   possible cases to best ensure correctness. This means not only maximizing
+   code coverage, but also covering all possible corner cases.
+
+2. Every pull request must pass all tests before it can be merged. The tests
+   are automatically run by the GitHub Actions CI on every pull request. If
+   any tests fail, the CI will fail with a red ❌. These failures must be
+   addressed before the pull request can be merged.
+
+3. Bug fixes should be accompanied by a [regression
+   test](writing-tests-regression-tests).
+
+(writing-tests-basics)=
+## Basics for Writing Tests
+
+Tests are located alongside the code in `tests/` directories, in files named
+`test_<thing>.py`. In most cases, if you modified
+`sympy/<submodule>/<file>.py` then the test for the functionality will go in
+`sympy/<submodule>/tests/test_<file>.py`. For example, the tests for the
+functions in `sympy/simplify/sqrtdenest.py` are in
+`sympy/simplify/tests/test_sqrtdenest.py`. There are some exceptions to this
+rule, so in general try to find where the existing tests are for a function
+and add your tests alongside them. If you are adding tests for a new function,
+follow the general pattern of tests in the module you are adding to.
+
+Tests follow a simple pattern, which should be apparent from reading the
+existing test files. Tests are in functions that start with `test_` and
+contain lines like
+
+```py
+assert function(arguments) == result
+```
+
+For example
+
+```py
+# from sympy/functions/elementary/tests/test_trigonometric.py
+
+def test_cos_series():
+    assert cos(x).series(x, 0, 9) == \
+        1 - x**2/2 + x**4/24 - x**6/720 + x**8/40320 + O(x**9)
+```
+
+New test cases can be added to an existing test function if it is relevant, or
+you can create a new test function.
+
+## Running Tests
+
+The basic way to run the tests is to use
+
+```
+./bin/test
+```
+
+to run the tests, and
+
+```
+./bin/doctest
+```
+
+to run the doctests. Note that the full test suite can take some time to run,
+so you may prefer to just run a subset of the tests, e.g., corresponding to
+the module you modified. You can do this by passing the name of the submodules
+or tests files to the test command. For example,
```

Comment:
I think "you may prefer" is a bit of an understatement. Unless I am making some deep changes I very rarely run the entire test suite and almost always just run the tests for a specific module or even submodule. The full test suite is far too slow to use during development iteration. I would rather explain this the other way around i.e. first suggest `bin/test solvers` and then only afterwards suggest that in some situations you might want to run the whole test suite.